### DISCUSSION TITLE
Multiple projects using the same namespace

### DISCUSSION THREAD

**Discussion by frederikprijck** [dotnet/docfx#8204] [Q&A]
Hello 👋 

I am trying to create a documentation site that gets build using multiple projects. To demonstrate the issue I am having, I simplified my setup to the following:

- ProjectA
  - [Class1.cs](https://github.com/frederikprijck/docfx-multiple-projects/blob/main/ProjectA/Class1.cs)
- ProjectB
  - [Class1.cs](https://github.com/frederikprijck/docfx-multiple-projects/blob/main/ProjectB/Class1.cs)

As you can see, both projects share the same namespace, and even has the same name for the class. Even though in reality both projects also have classes unique for that project, there are lot of classes that use the exact same name and namespace.

The use-case here is that we have a couple of platform specific SDKs that share the same API for the most part (think Xamarin, WinForms, WPF, ...). Preferably, I would want to keep the structure of the project and its namespacing the same as it is today, as it would be a breaking change for us to be changing these around.

With the above structure, and the docfx.json as seen here: https://github.com/frederikprijck/docfx-multiple-projects/blob/main/docs-source/docfx.json, we noticed the output `toc.html` is [using an incorrect `href`](https://github.com/frederikprijck/docfx-multiple-projects/blob/main/docs/toc.html#L14-L21): 

```html
<ul class="nav level1">
  <li>
    <a href="api-a/Project.html" name="api-a/toc.html" title="A">A</a>
  </li>
  <li>
    <a href="api-a/Project.html" name="api-b/toc.html" title="B">B</a>
  </li>
</ul>
```

As you can see, the `name` is correct, but the `href` uses `api-a` for both `A` and `B`.
Even though the `toc.html` is wrong, the page `api-b/Project.html` does exist, as you can see here: https://github.com/frederikprijck/docfx-multiple-projects/blob/main/docs/api-b/Project.html.

Additionally, I noticed that links in `api-b/Project.html` can point to `api-a` when the type in question exists in api-a as well, e.g.: https://github.com/frederikprijck/docfx-multiple-projects/blob/main/docs/api-b/Project.html#L80. I would have expected this link to point to [`api-b/Project.class1.html`](https://github.com/frederikprijck/docfx-multiple-projects/blob/main/docs/api-b/Project.Class1.html) instead.

Would anyone be aware of anything I am doing wrong that results in the incorrect rendering, or is using the same namespace and class name in multiple project not something that's supposed to be supported?

Possibly related: https://github.com/dotnet/docfx/discussions/8018

Thanks!

**Comment by tcc-mark** [comment]
I'm having this same issue too, does anyone know if a solution to this exists?

**Reply by MaceWindu** [reply]
add `"globalNamespaceId": "your_unique_discriminator_here"` to `metadata[]` entries

**Reply by tcc-mark** [reply]
Thanks, that does sound like what I'm looking for, but as far as I can see after adding that and rebuilding nothing has changed. For reference, my metadata section in docfx.json essentially looks like this:

``` 
"metadata": [
    {
      "src": [
        {
          "files": [
            "path/to/projectA.sln"
          ]
        }
      ],
      "dest": "api/projectA",
      "globalNamespaceId": "ProjectA"
    },
    {
      "src": [
        {
          "files": [
            "path/to/projectB.sln"
          ],
        }
      ],
      "dest": "api/projectB",
      "globalNamespaceId": "ProjectB"
    }
  ]
```

**Reply by MaceWindu** [reply]
Strange, in our case it helped. No idea why it doesn't work, maybe you can find some hints in our config https://github.com/linq2db/docs/blob/master/source/docfx.json

**Reply by tcc-mark** [reply]
I unfortunately had no luck trying to get it working using your config as a reference. I then had a look to see how it works in the docs for your project, and it looks like you currently have the same issue too? (on this page: https://linq2db.github.io/index.html). In the main Linq To DB project, clicking on the LinqToDB namespace at the top of the navigation takes me to that namespace within the Tools project.

Unless a fix just hasn't been applied to your live docs yet?

**Reply by MaceWindu** [reply]
whoops, you are right. It fixed this issue https://github.com/linq2db/linq2db/issues/4141 and I had impression everything is working fine now. Will look into it once more
